lib/contract/validateDPContractFactory.js

```diff
@@ -20,10 +24,75 @@ module.exports = function validateDPContractFactory(validator) {
     // TODO: Use validateSchema
     //  https://github.com/epoberezkin/ajv#validateschemaobject-schema---boolean
 
-    return validator.validate(
+    const result = validator.validate(
       JsonSchemaValidator.SCHEMAS.META.DP_CONTRACT,
       rawDPContract,
     );
+
+    if (!result.isValid()) {
+      return result;
+    }
+
+    // Validate indices
+    Object.entries(rawDPContract.dpObjectsDefinition).filter(([, dpObjectDefinition]) => (
+      Object.prototype.hasOwnProperty.call(dpObjectDefinition, 'indices')
+    ))
+      .forEach(([dpObjectType, dpObjectDefinition]) => {
+        const indicesFingerprints = [];
+
+        dpObjectDefinition.indices.forEach((indexDefinition) => {
+          const indexPropertyNames = Object.keys(indexDefinition.properties);
+
+          const indicesFingerprint = JSON.stringify(indexDefinition.properties);
+
+          // Ensure index definition uniqueness
+          if (indicesFingerprints.includes(indicesFingerprint)) {
+            result.addError(
+              new DuplicateIndexError(
+                rawDPContract,
+                dpObjectType,
+                indexDefinition,
+              ),
+            );
+          }
+
+          indicesFingerprints.push(indicesFingerprint);
+
+          // Currently, only user-based MN quorums are implemented
```

Comment:
Actually none of them working as at the time of writing, but DPA quorum shouldn't be different from the user quorum. The only difference is that quorum is selected based on the DPA id instead of user id